### PR TITLE
Batch editing

### PR DESCRIPTION
**Pull request by toban**

### CODE REVIEW

**Comment by toban** [comment]
- [x] savedOtherItems might reference not needed
- [x] No exception add break
- [x] Add result only if status is good or ok
- [ ] change user to private on EditEndpoint
- [ ] persistItem can set the item revision instead of returning saved items.
- [x] just return an array on parseBatchRequests - remove BatchEditRequest
- [x] Add PHP integration test for batch-edit endpoint

File: includes/MediaWiki/Api/EditEndpoint.php [hunk 1]
```diff
@@ -52,69 +53,31 @@ public function __construct(
...
-			$reconciledItem = $this->itemReconciler->reconcileItem(
-				$inputEntity,
-				$reconcileUrlProperty
-			);
-		} catch ( ReconciliationException $rex ) {
-			throw new LocalizedHttpException( $rex->getMessageValue(), 400 );
-		}
-
-		// And make the edit
+	public function persistItem( ReconciledItem $reconciledItem, array $otherItems ): array {
```
Comment (toban) [inline includes/MediaWiki/Api/EditEndpoint.php:83]: Add PHPDOC

Comment (lucaswerkmeister) [reply includes/MediaWiki/Api/EditEndpoint.php:83]: Done

**Comment by lucaswerkmeister** [comment]
> just return an array on parseBatchRequests - remove BatchEditRequest

done, but all of the tests still passed even though I hadn’t updated them, which means

- [x] add tests for parseBatchRequests

**Comment by lucaswerkmeister** [comment]
on the other points:

> change user to private on EditEndpoint

not convinced this one is needed tbh. I’d be willing to merge without this

> persistItem can set the item revision instead of returning saved items.

to be done in a follow-up

**Review by lucaswerkmeister — APPROVED** [review]

**Comment by guergana** [comment]
LGTM. Also nice now we new tests.

File: includes/MediaWiki/Api/BatchEditEndpoint.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			$response['results'] = array_map( static function ( $result ) {
+				/** @var EntityRevision $entityRevision */
+				$entityRevision = $result['revision'];
+				return [
+					'entityId' => $entityRevision->getEntity()->getId()->getSerialization(),
+					'revisionId' => $entityRevision->getRevisionId(),
+				];
+			}, $status->getValue() );
+		}
+		// TODO attach errors
```
Comment (toban) [inline includes/MediaWiki/Api/BatchEditEndpoint.php:101]: We could do this now maybe or write this down as a ticket? I suspect there might be a series of followups planned already?

File: tests/phpunit/integration/MediaWiki/Api/BatchEditEndpointTest.php [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Snak\PropertyValueSnak;
+use Wikibase\Repo\WikibaseRepo;
+
+/**
+ * @covers MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Api\BatchEditEndpoint
+ * @group Database
+ * @license GPL-2.0-or-later
+ */
+class BatchEditEndpointTest extends \MediaWikiIntegrationTestCase {
```
Comment (toban) [inline tests/phpunit/integration/MediaWiki/Api/BatchEditEndpointTest.php:28]: This reminds me a lot of EditEndpoint test, not sure there are any real benefits of doing so but maybe we could refactor these afterwards to share some of this code / token tests etc if we feel there is enough time.

File: tests/phpunit/unit/MediaWiki/Request/EditRequestParserTest.php [hunk 4]
```diff
@@ -97,7 +97,7 @@ private function getPropertyDataTypeLookup(): PropertyDataTypeLookup {
 		return $propertyDataTypeLookup;
 	}
 
-	public function testParseRequestInterface_good(): void {
+	public function testParseRequestBody_good(): void {
```
Comment (toban) [inline tests/phpunit/unit/MediaWiki/Request/EditRequestParserTest.php:100]: good catch

**Review by toban — COMMENTED** [review]
Can't approve this but I do! it looks great. Just some comments for possible followups.

File: includes/MediaWiki/Api/BatchEditEndpoint.php (hunk 2 above)
Comment (lucaswerkmeister) [reply includes/MediaWiki/Api/BatchEditEndpoint.php:101]: Ticket makes sense, I think. (The same comment applies to `SingleEditEndpoint`, though we don’t have a TODO there.)

Comment (toban) [reply includes/MediaWiki/Api/BatchEditEndpoint.php:101]: https://phabricator.wikimedia.org/T284134

**Pull request merged**